exchanges/binance/binance_wrapper.go

```diff
@@ -1793,3 +1803,68 @@ func (b *Binance) GetAvailableTransferChains(ctx context.Context, cryptocurrency
 	}
 	return availableChains, nil
 }
+
+// GetEnabledPairs is a method that returns the enabled currency pairs of
+// the exchange by asset type, if the asset type is disabled this will return no
+// enabled pairs
+// overrides default implementation to use optional delimiter
+func (b *Binance) GetEnabledPairs(a asset.Item) (currency.Pairs, error) {
+	err := b.CurrencyPairs.IsAssetEnabled(a)
+	if err != nil {
+		return nil, nil // nolint:nilerr // non-fatal error
+	}
+	format, err := b.GetPairFormat(a, false)
+	if err != nil {
+		return nil, err
+	}
+	enabledPairs, err := b.CurrencyPairs.GetPairs(a, true)
+	if err != nil {
+		return nil, err
+	}
+	if a == asset.USDTMarginedFutures {
+		var resp currency.Pairs
+		for i := range enabledPairs {
+			pair := enabledPairs[i]
+			if pair.Delimiter == currency.UnderscoreDelimiter {
+				// we cannot remove expiring contract delimiters from requests
+				pair = pair.Format(currency.UnderscoreDelimiter, format.Uppercase)
+			} else {
+				pair = pair.Format(format.Delimiter, format.Uppercase)
+			}
+			resp = append(resp, pair)
+		}
+		return resp, nil
+	}
+
+	return enabledPairs.Format(format.Delimiter,
+			format.Index,
+			format.Uppercase),
+		nil
```

Comment:
```go
	enabledPairs, err := b.CurrencyPairs.GetPairs(a, true)
	if err != nil {
		return nil, err
	}
	format, err := b.GetPairFormat(a, false)
	if err != nil {
		return nil, err
	}

	fmted := enabledPairs.Format(format.Delimiter, format.Index, format.Uppercase)
	if a == asset.USDTMarginedFutures {
		for i := range fmted {
			if fmted[i].Delimiter == currency.UnderscoreDelimiter {
				// we cannot remove expiring contract delimiters from requests
				fmted[i].Delimiter = currency.UnderscoreDelimiter
			}
		}
	}
	return fmted, nil
```

Format selected and then add the delimiter.

Reply:
That would remove the delimiter, there'd be no way of knowing if there was an underscore delimiter to keep it. The issue is that for USDTMarginedFutures, we need to have an optional delimiter for the asset type. For all other USDTMarginedFutures pairs, we don't have a delimiter and Binance would reject it. For the specific currencies that do, it requires the underscore.

It does highlight that my test `TestGetEnabledPairs` wasn't adequate enough as it passes despite this change, but the currency has no delimiter

Reply:
👍